lib/fetching/content-services/facebook-content-service.js

```diff
@@ -26,116 +27,139 @@ FacebookContentService.prototype._doFetch = function(options, callback) {
-// (does not need to be sorted, parent class handles that).
-FacebookContentService.prototype._handleResults = function(posts, comments, authors) {
+FacebookContentService.prototype._handleResults = function(data, source) {
   var self = this;
+  var oldPost = this.posts[source];
   var reportData = [];
-  var postsById = _.indexBy(posts, 'post_id');
 
-  // Handle each post.
-  posts.forEach(function(post) {
-    reportData.push(self._parse(post, {type: 'post', authors: authors}));
-  });
+  if (!data) {
+    return reportData;
+  }
 
-  // Handle each comment.
-  comments.forEach(function(comment) {
-    reportData.push(self._parse(comment, {type: 'comment', authors: authors, post: postsById[comment.post_id]}));
+  data.forEach(function(object) {
+    //If object has a type then it is a post, else it is a comment
+    if (object.type) {
+      // Handle each post and any comment already attached to it
+      reportData.push(self._parse(object, {type: 'post'}));
+      if (object.comments && object.comments.data && object.comments.data.length > 0 ) {
+        object.comments.data.forEach(function(comment) {
+          reportData.push(self._parse(comment, {type: 'comment', url: object.link}));
+        });
+      }
+    } else {
+      if (oldPost == undefined){
+        reportData.push(self._parse(object, {type: 'comment', url: object.link}));
+      } else {
+        reportData.push(self._parse(object, {type: 'comment', url: oldPost.link }));
+      }
+    }
   });
 
-  // Remove any that are too old.
-  // (Remember, we are grabbing posts by updated_time to get at their comments, but we may have already stored them).
+  //Store most recent post id to retrieve post comments
+  if (data[0] && data[0].type && data[0].id) {
+    this.posts[source] = {
+      id: data[0].id,
+      link: data[0].link
+    };
+  }
+
+  //Filter out old data if any
   if (self._lastReportDate)
     reportData = _.filter(reportData, function(rd) { return rd.authoredAt > self._lastReportDate });
 
   return reportData;
 };
 
-// Parse each fql result into our data format
 FacebookContentService.prototype._parse = function(data, options) {
-  switch (options.type) {
+
+  switch(options.type) {
     case 'post':
       return {
-        authoredAt: new Date(data.created_time * 1000),
+        authoredAt: new Date(data.created_time),
         fetchedAt: new Date(),
         content: data.message,
-        author: this._findAuthor(data.actor_id, options.authors),
-        url: data.permalink
+        author: data.from.name,
+        url: data.link
       };
     case 'comment':
       return {
-        authoredAt: new Date(data.time * 1000),
+        authoredAt: new Date(data.created_time),
         fetchedAt: new Date(),
-        content: data.text,
-        author: this._findAuthor(data.fromid, options.authors),
-        url: options.post.permalink
+        content: data.message,
+        author: data.from.name,
```

Comment:
I have now got a 2.0 key and got an instance of aggie graph_api running with 13 different sources. Unsurprisingly the limit was reached quite early. 

More interestingly some of the reports (photos, links,...) have come back with empty content which breaks the angular app because of the Autolinker call

```
var linkify = function(report) {
      report.content = Autolinker.dulink(report.content);
      return report;
    };

```